python_modules/dagster/dagster/serdes/serdes.py

```diff
@@ -364,17 +358,21 @@ def _unpack_value(val: Any, whitelist_map: WhitelistMap, descent_path: str) -> A
                 f"expected versions.{_path_msg(descent_path)}"
             )
 
-        klass, serializer, sig_params = whitelist_map.get_tuple_entry(klass_name)
+        klass, serializer, args_for_class = whitelist_map.get_tuple_entry(klass_name)
 
         # Target class being set to none, likely by
         if klass is None:
             return None
 
+        # Naively implements backwards compatibility by filtering arguments that aren't present in
+        # the constructor. If a property is present in the serialized object, but doesn't exist in
+        # the version of the class loaded into memory, that property will be completely ignored.
```

Comment:
But this limits the values that can be considered from the `value_from_storage_dict` function, right?

I'm surprised we don't have serializers that use this functionality (converting old attributes to new attributes)

Reply:
hmm yea this probably messes with `pipeline_run_from_storage` in ways that just are not under test

Reply:
yeah, I think we want a pre-unpack filter, probably another method on the serializer (e.g. `should_unpack` that takes in the key, klass, and args_for_class)

Reply:
ok respun with this in mind, did the try-hard version of refactoring instead of the `bool` approach